javascript/javascript2/week1/index.html

```diff
@@ -0,0 +1,120 @@
+<!DOCTYPE html>
+<html lang=="EN">
+
+<head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" type="text/css" href="styles.css" />
+    <title>spirit animal</title>
+
+
+</head>
+
+
+<body>
+
+
+    <div id="maindiv">
+
+        <h1>GENERATING SPIRIT ANIMAL</h1>
+        <div>
+            <label for="optional">Choose when you want to generate animal</label>
+
+
+            <select id="selection" class="field" name="options">
+  <option value="clicked">On button click</option>
+  <option value="hovered">Hover input</option>
+  <option value="written">Text written in field</option>
+
+            </select>
+        </div>
+        <div>
+            <label for="txt">Insert your name: </label>
+
+            <input id="myIn" class=field type="text" />
+
+            <button id="mybtn">Generate</button>
+
+            <div id='div2'> </div>
+
+        </div>
+    </div>
+
+    <script>
+        let animals = [
+            "Flamingo ",
+            "Cheetah ",
+            "Owl ",
+            "Sheep ",
+            "Lion ",
+            "Reindeer ",
+            "Swan ",
+            "Butterfly ",
+            "Dog ",
+            "Dolphin ",
+        ];
+
+        function getSelect(sel) {
+            let value = sel.options[sel.selectedIndex].value;
+            return value;
+
+        }
+
+        let select = document.getElementById('selection');
+
+        let btn = document.getElementById("mybtn");
+        let myinput = document.getElementById("myIn");
+
+        //value.onchange = getSelect(select);
+
+
+
+        function genAnimal(a) {
+            let yourAnimal = a[Math.floor(Math.random() * a.length)];
+
+
+
+            let yourName = myinput.value;
+
+            let i;
+            let node = document.createElement("div");
+            let textnode = document.createTextNode(i);
+            node.appendChild(textnode);
+            document.querySelector("#maindiv").appendChild(node).innerHTML = yourName + "-" + yourAnimal;;
+
+            node.style.backgroundColor = "violet";
+            node.style.display = "flex";
+        };
+
+
+
+
+        if (getSelect(select) === "clicked") {
+            btn.addEventListener("click", function() {
+
+                genAnimal(animals);
+
+            });
+        }
+        if (getSelect(select) === "hovered") {
+
+            myinput.addEventListener("mouseover", function(event) {
+
+                event.target.style.backgroundColor = "yellow";
+                genAnimal(animals);
+
+            });
+        }
+        if (getSelect(select) === "written") {
+            myinput.addEventListener("input", function(event) {
+
+                event.target.style.backgroundColor = "pink";
+                genAnimal(animals);
+
+            });
+        }
```

Comment:
There should be a way of picking correct option from select, in order to change "handle" method.
You could try apply addEventListener to select element with "change" event.
let select = document.getElementById('selection');

and based on the value e.target.value, add new eventListenerS

But you also need to remember to remove other event listeners. 
Fx. when you use click, you don't need to have hover or input eventlisteners.

Reply:
![getAnimalexercise](https://user-images.githubusercontent.com/81299490/130069530-8d05d133-edbf-41b6-ac63-820384cbb18e.PNG)

I thought that problem with removing other event listeners will solve itselfs with IF statement, but it did not. Then I tried removeEventListener(),and everything stopped to work :/ Can you suggest please how should they be removed? xD @OlkaA

Reply:
Good question 👍 
I think you need to add second parameter to the removeEventListener(), otherwise it gives error in the console.
From documentation 
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener
![image](https://user-images.githubusercontent.com/31403464/130133723-daf99f82-01a2-49f4-b605-496531d842e7.png)
Example:
![image](https://user-images.githubusercontent.com/31403464/130133756-c84b66f2-f288-436e-87f9-e51e7bf4720f.png)

Works for me like this (quick solution, might be improved): 
```
// for initial (first value in select,  :) )
btn.addEventListener("click", genAnimal);

//change event of select
select.addEventListener("change", (event) => {
    if (event.target.value === "clicked") {            
            myinput.removeEventListener("mouseover", genAnimal);
            myinput.removeEventListener("input", genAnimal);
            btn.addEventListener("click", genAnimal);
        }
        if (event.target.value === "hovered") {
            myinput.removeEventListener("input", genAnimal);
            btn.removeEventListener("click", genAnimal);
            myinput.addEventListener("mouseover", genAnimal);
        }
        if (event.target.value === "written") {
            myinput.removeEventListener("mouseover", genAnimal);
            btn.removeEventListener("click", genAnimal);
            myinput.addEventListener("input", genAnimal);
        }
})

//also get Animal function should be modified:
function genAnimal() {
            let yourAnimal = animals[Math.floor(Math.random() * animals.length)];
```